### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Rilton**
Hi everyone,

First of all, thanks for the QuickStart, they helped a lot in my current engagement.

Second, I believe that this is a good time in my career starting giving back to the community, hence thought others my find  helpful to have this  "Quickstart #9:  MongoDB configuration".

Aiming to keep this "Quickstart #9" sample  close as possible to others  QuickStart samples, all MongoDB required changes are based on "IdentityServerBuilderExtensions" and properly wired in Startup.ConfigureServices(). 

Also, you may notice that all related readme.md files were updated.

And finally, if you identify anything that you would like to have it enhanced or refactored in order to accept this small contribution just give me a shout. 

Thanks again,

### CODE REVIEW

**Comment by AlexZeitler** [comment]
Tested it using a MongoDb instance running in Docker on macOS - works.

**Comment by Rilton** [comment]
Awesome! Thanks

**Comment by brockallen** [comment]
@leastprivilege thoughts on this? personally I'm inclined to no accept this here, and rather post it as a link to @Rilton's repo. Or we can put it elsewhere in the samples, but not as part of the quickstarts, imo.

**Comment by AlexZeitler** [comment]
Some projects handle this having a contrib folder or contrib project.

**Comment by brockallen** [comment]
Yea, don't get me wrong @Rilton and @AlexZeitler -- I'm all for it, but as a quickstart that's a bit more formal and would require us to maintain it (and I'm not a mongo programmer).

**Comment by AlexZeitler** [comment]
Yeah, totally understand this - OTOH people might be interested in stuff you can't provide on your own as well (like Mongo).

**Comment by Rilton** [comment]
@AlexZeitler and @leastprivilege, I understand your perspective and concern, and I know I am new here but if you guys want me to be responsible for supporting that particular sample (or any other) I will be more than happy to contribute.

If you wish me to organize the project in another format in order to have it under some form of contrib repo just let me know.

Also in order to save some time,  I created a new repo just with the Mongo project sample in case you decide to just have link to my repo -  the link to that new repo is: https://github.com/Rilton/IdentityServer4.Samples.Mongo

Bottom line, I really would like to be able to assist this project and help this community.

**Comment by leastprivilege** [comment]
Thanks for the contribution.

How about this - you should have a separate repo with the walkthrough instructions, code starting point and reference solution. We will create a new section in the quickstarts for external contributions and link to it.

We are already extremely swamped with keeping track of our own content and it is not feasible anymore to add external content into our own repos. I hope you understand.

**Comment by EdwardBlair** [comment]
@Rilton et al.
I made this https://github.com/EdwardBlair/IdentityServer4.MongoDBDriver last week to improve upon [this](https://github.com/diogodamiani/IdentityServer4.MongoDB). I believe it has a few issues, of which I won't go in to detail in here, that I have addressed. *N.B. The name needs changing as per [this](https://identityserver4.readthedocs.io/en/release/intro/contributing.html#contribution-projects) so if anyone has any good ideas, please don't be shy!*

It has a familiar interface -- not dissimilar to that which `IdentityServer4.EntityFramework` uses -- so writing a guide using this library would be make for a more consistent guide.

I believe rather than having a "built-in" implementation of IClientStore etc.. for MongoDB, depending on an external persistence layer would be beneficial. This way it can be improved upon externally and used straight away in production *(famous last words)*, rather than providing stepping stone tutorial code which naive persons may use sans modification.

What are your thoughts?

**Comment by brockallen** [comment]
We'll close this, and can you team up with @EdwardBlair and let us know where your sample end up. Thanks!

**Comment by Rilton** [comment]
Not problem. I will work with  @EdwardBlair.
